src/App.jsx

```diff
@@ -14,45 +17,88 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button
-        type="button"
-        className="button is-info is-light"
-      >
-        Sort alphabetically
-      </button>
-
-      <button
-        type="button"
-        className="button is-success is-light"
-      >
-        Sort by length
-      </button>
-
-      <button
-        type="button"
-        className="button is-warning is-light"
-      >
-        Reverse
-      </button>
-
-      <button
-        type="button"
-        className="button is-danger is-light"
-      >
-        Reset
-      </button>
-    </div>
+const SORT_ALPHABETICAL = 'alphabetical';
+const SORT_BY_LENGTH = 'length';
+const SORT_BY_DEFAULT = '';
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
+const GoodList = ({ goods }) => (
+  <ul>
+    {goods.map(good => (
+      <li data-cy="Good">
+        {good}
+      </li>
+    ))}
+  </ul>
 );
+
+export const App = () => {
+  const [sortField, setSortField] = useState('');
+  const [isReversed, setIsReserved] = useState(false);
+  const visibleGoods = [...goodsFromServer].sort((good1, good2) => {
+    switch (sortField) {
+      case SORT_ALPHABETICAL:
+        return good1.localeCompare(good2);
+
+      case SORT_BY_LENGTH:
+        return good1.length - good2.length;
+
+      default:
+        return 0;
+    }
+  });
+
+  if (isReversed) {
+    visibleGoods.reverse();
+  }
```

Comment:
move sorting method to an own helper function. 
And here just invoke this function like
```suggestion
  const visibleGoods = getSortedGood(goods, filters)
```